frontend/catalyst/utils/toml.py

```diff
@@ -52,25 +55,65 @@ def read_toml_file(toml_file: str) -> TOMLDocument:
     return config
 
 
-def check_mid_circuit_measurement_flag(config: TOMLDocument) -> bool:
-    """Check the global mid-circuit measurement flag"""
-    return bool(config.get("compilation", {}).get("mid_circuit_measurement", False))
+@dataclass(unsafe_hash=True)
+class OperationProperties:
+    """Capabilities of a single operation"""
 
+    invertible: bool
+    controllable: bool
+    differentiable: bool
 
-def check_adjoint_flag(config: TOMLDocument, shots_present: bool) -> bool:
-    """Check the global adjoint flag for toml schema 1. For newer schemas the adjoint flag is
-    defined to be set if all native gates are inverible"""
-    schema = int(config["schema"])
-    if schema == 1:
```

Comment:
Do we give up the support for schema 1?

Reply:
No, we just move the toml parsing procedure into the `toml.py`, see [here](https://github.com/PennyLaneAI/catalyst/blob/c803ecf664bff686843b0919af1223c5a1498d1c/frontend/catalyst/utils/toml.py#L262).
The idea is to encode capabilities with our own data structures (the DeviceConfig dataclass) and use it in our calculations independently of the serialization format. 
The PL/L toml schema 2 is not merged yet. In fact we mostly work with schema 1 now for main devices.